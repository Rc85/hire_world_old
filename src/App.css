.panel-container {
    display: flex;
    justify-content: center;
}

.main-container {
    flex: 1;
    position: relative;
    display: flex;
    justify-content: center;
}

.blue-panel {
    background: #2F496E;
}

.blue-panel.deep {
    padding: 100px;
}

.blue-panel.shallow {
    padding: 25px;
}

.blue-panel.rounded-corners {
    border-radius: 5px;
}

.form-buttons {
    text-align: right;
}

.form-buttons input[type=submit] {
    margin-right: 5px;
}

.section-container {
    margin-bottom: 20px;
}

.main-panel {
    padding: 100px 200px;
    display: flex;
    flex-direction: column;
}

.three-rounded {
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    border-top-right-radius: 5px;
}

.section-container .section-header {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    background: linear-gradient(#406497, #2F496E);
    padding: 15px;
}

.section-container .section-body {
    border-width: 0 1px 1px 1px;
    border-color: #2F496E;
    border-style: solid;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    padding: 15px;
    color: #000;
}

.list-horizontal {
    display: flex;
    flex-wrap: wrap;
}

.col-container {
    display: flex;
    flex-direction: column;
    width: 100%;
    min-height: 100%;
    position: relative;
}

.row-container {
    display: flex;
    justify-content: space-between;
    width: 100%;
    min-height: 100%;
    flex-wrap: wrap;
}

.section-container {
    width: 17%;
}

.submit-btn {
    width: 90px;
}

.menu-container {
    position: relative;
}

.menu {
    position: absolute;
    background: #C4C4C4;
    z-index: 1;
}

.menu-button {
    cursor: pointer;
}

.menu-item {
    padding: 10px;
    min-width: 75px;
    text-align: center;
    color: #000;
}

.menu-item:hover {
    background: #2988BC;
    cursor: pointer;
    color: #FFF;
}

.menu-item:last-child {
    border: 1px solid #2988BC;
}

.menu-item:not(:last-child) {
    border-width: 1px 1px 0 1px;
    border-color: #2988BC;
    border-style: solid;
}

.tab-bar {
    display: flex;
}

.tab-button {
    border-top-right-radius: 5px;
    border-top-left-radius: 5px;
    border-width: 1px 1px 0 1px;
    border-color: #2F496E;
    border-style: solid;
    padding: 10px;
    color: #000;
    margin-right: 5px;
}

.tab-button:hover {
    background: #2988BC;
    cursor: pointer;
    color: #F9F6F4;
}

.tab-button.active {
    background: #2F496E;
    color: #F9F6F4;
}

.delete-button {
    cursor: pointer;
}

.required-asterisk {
    color: #ED463D;
}

.grey-panel {
    background: #F9F6F4;
    padding: 10px;
    color: #000;
}