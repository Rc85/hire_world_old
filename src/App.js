import React, { Component } from 'react';
import { Route, withRouter, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { GetSession, GetSectors } from './actions/FetchActions';
import { RemoveAlert } from './actions/AlertActions';
import * as Pages from './components/pages';
import * as Admin from './components/admin';
import TopBar from './components/includes/site/TopBar';
import BrowseMenu from './components/includes/site/BrowseMenu';
import Confirmation from './components/utils/Confirmation';
import Alert from './components/utils/Alert';
import Prompt from './components/utils/Prompt';
import Warning from './components/utils/Warning';
import { ToggleMenu } from './actions/MenuActions';

class App extends Component {
	constructor(props) {
		super(props);

		this.state = {
			mainMenu: false
		}
	}

	componentDidUpdate(prevProps, prevState) {
		if (prevProps.location.key !== this.props.location.key) {
			this.props.dispatch(GetSession());
			this.props.dispatch(GetSectors());
		}
	}

	componentDidMount() {
		this.props.dispatch(GetSession());
		this.props.dispatch(GetSectors());

		document.body.addEventListener('click', (e) => {
			if (typeof e.target.className === 'object' || e.target.classList.contains('admin-menu-button') || e.target.classList.contains('menu-item') || e.target.classList.contains('menu') || e.target.id === 'browse-menu-button') {
				return;
			} else if (this.props.menu.open) {
				this.props.dispatch(ToggleMenu('', ''));
			}
		});
    }

	render() {
		let confirmation, sectors, alerts, prompt, warning;

		if (this.props.confirmation.status === true) {
			confirmation = <Confirmation message={this.props.confirmation.message} note={this.props.confirmation.note} />
		}

		if (this.props.sectors) {
			sectors = this.props.sectors.map((sector, i) => {
				return <Route key={i} path={`/sectors/${sector.sector.toLowerCase()}`} render={() => <Pages.Sectors name={sector.sector} />} />;
			});
		}

		if (this.props.menu.open === 'main') {
			this.menu = <BrowseMenu sectors={this.props.sectors} hide={false} />
		} else {
			this.menu = <BrowseMenu sectors={this.props.sectors} hide={true} />
		}

		if (this.props.alerts.length > 0) {
			alerts = this.props.alerts.map((alert, i) => {
				return <Alert key={i} status={alert.status} message={alert.message} unmount={() => {
					this.props.alerts.splice(0, 1);
					this.props.dispatch(RemoveAlert(this.props.alerts))
				}} />
			});
		} else if (this.props.alerts.length === 0) {
			alerts = [];
		}

		if (this.props.prompt.text) {
			prompt = <Prompt text={this.props.prompt.text} data={this.props.prompt.data} />;
		}

		if (this.props.warning.message) {
			warning = <Warning message={this.props.warning.message} />;
		}

		return (
			<div className='col-container'>
				{warning}
				{prompt}
				{confirmation}
				<TopBar />

				<div className='position-relative'>{this.menu}</div>

				<section className='main-container'>
					<Switch>
						<Route exact path='/' component={Pages.Main} />
						<Route exact path='/view' component={Pages.ViewUser} />
						<Route exact path='/dashboard/list' render={() => <Pages.Dashboard user={this.props.user}><Pages.ListSettings user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/saved_listings' render={() => <Pages.Dashboard user={this.props.user}><Pages.SavedListings user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/edit' render={() => <Pages.Dashboard user={this.props.user}><Pages.EditUser user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/settings' render={() => <Pages.Dashboard user={this.props.user}><Pages.UserSettings user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/dashboard/messages/:stage' render={() => <Pages.Dashboard user={this.props.user}><Pages.Messages user={this.props.user} /></Pages.Dashboard>} />} />
						<Route exact path='/dashboard/message/:stage/:id/details' render={() => <Pages.Dashboard user={this.props.user}><Pages.MessageDetails user={this.props.user} /></Pages.Dashboard>} />
						<Route exact path='/listing/:id' render={() => <Pages.ListingDetails user={this.props.user} />} />
						<Route exact path='/user/:username' render={() => <Pages.ViewUser user={this.props.user} />} />
						<Route exact path='/account/login' render={() => <Pages.Login user={this.props.user} />} />
						<Route exact path='/account/register' render={() => <Pages.Register user={this.props.user} />} />
						{sectors}
						<Route exact path='/sectors/:sector' component={Pages.Sectors} />

						<Route exact path='/admin-panel' render={() => <Admin.Admin><Admin.AdminOverview user={this.props.user} /></Admin.Admin>} />
						<Route exact path='/admin-panel/sectors' render={() => <Admin.Admin><Admin.AdminSectors user={this.props.user} /></Admin.Admin>} />
						<Route exact path='/admin-panel/users' render={() => <Admin.Admin><Admin.AdminUsers user={this.props.user} /></Admin.Admin>} />
						<Route exact path='/admin-panel/listings' render={() => <Admin.Admin><Admin.AdminListings user={this.props.user} sectors={this.props.sectors} /></Admin.Admin>} />

						<Route render={() => <Pages.Response code={404} header={'Not Found'} message={`This page you're trying to access does not exist.`} />} />
					</Switch>
				</section>

				<div className='alert-container'>{alerts}</div>
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		user: state.Login,
		confirmation: state.Confirmation,
		sectors: state.Sectors.sectors,
		alerts: state.Alert.alerts,
		prompt: state.Prompt,
		warning: state.Warning,
		menu: state.Menu
	}
}

export default withRouter(connect(mapStateToProps)(App));
