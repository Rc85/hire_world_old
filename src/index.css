@font-face {
    font-family: 'Montserrat';
    src: url('./fonts/Montserrat-Regular.ttf');
}

@font-face {
    font-family: 'Montserrat Bold';
    src: url('./fonts/Montserrat-Bold.ttf');
}

html {
    min-height: 100vh;
}

body {
    margin: 0;
    padding: 0;
    font-family: 'Montserrat';
    background: #C4C4C4;
    color: #F9F6F4;
    min-height: 100vh;
    display: flex;
    overflow-x: hidden;
}

#root {
    width: 100%;
    min-height: 100%;
}

a {
    font-weight: bold;
    color: #F9F6F4;
}

a:hover {
    color: #F9F6F4;
    text-decoration: underline;
}

h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat Bold';
}

hr {
    background: #2988BC;
}

/* Widths */

.w-5 {
    width: 5%;
}

.w-10 {
    width: 10%;
}

.w-20 {
    width: 20%;
}

.w-30 {
    width: 30%;
}

.w-33 {
    width: 33%;
}

.w-35 {
    width: 35%;
}

.w-40 {
    width: 40%;
}

.w-45 {
    width: 45%;
}

.w-55 {
    width: 55%;
}

.w-60 {
    width: 60%;
}
