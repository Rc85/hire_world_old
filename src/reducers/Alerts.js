const initialState = {
    alerts: []
};

export const Alert = (state = initialState, action) => {
    switch(action.type) {
        case 'ADD_ALERT': return Object.assign({}, state, {alerts: [...state.alerts, {status: action.status, message: action.message}]});
        case 'REMOVE_ALERT': return Object.assign({}, state, {alerts: [...action.alerts]});
        default: return state;
    }
}