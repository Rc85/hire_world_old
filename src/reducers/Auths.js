const loginInitialState = {
    status: '',
    user: null
}

export const Login = (state = loginInitialState, action) => {
    switch(action.type) {
        case 'LOGIN_USER_BEGIN':
        case 'LOGIN_USER_FAIL':
        case 'LOGIN_USER_ERROR':
            return Object.assign(...state, {status: action.status});
        case 'LOGIN_USER_SUCCESS':
        case 'EDIT_USER_SUCCESS':
        case 'GET_SESSION_SUCCESS':
        case 'GET_SESSION_FAIL':
        case 'EDIT_USER_BEGIN':
        case 'EDIT_USER_FAIL':
        case 'EDIT_USER_ERROR':
        case 'SAVE_LOCATIONS_SUCCESS':
        case 'SAVE_LOCATIONS_BEGIN':
        case 'SAVE_LOCATIONS_FAIL':
        case 'SAVE_LOCATIONS_ERROR':
        case 'SAVE_PROFILE_SUCCESS':
        case 'SAVE_PROFILE_BEGIN':
        case 'SAVE_PROFILE_FAIL':
        case 'SAVE_PROFILE_ERROR':
        case 'SAVE_EMAIL_SUCCESS':
        case 'SAVE_EMAIL_BEGIN':
        case 'SAVE_EMAIL_FAIL':
        case 'SAVE_EMAIL_ERROR':
            return Object.assign(...state, {status: action.status, user: action.user});
        case 'RESET_STATUS': return Object.assign(...state, {status: '', user: action.user});
        case 'LOGOUT_USER':
        default: return state;
    }
}

const registerInitialState = {
    status: null
}

export const Register = (state = registerInitialState, action) => {
    switch(action.type) {
        case 'REGISTER_USER_BEGIN':
        case 'REGISTER_USER_FAILED':
            return Object.assign(...state, {status: action.status});
        case 'REGISTER_USER_SUCCESS': return Object.assign(...state, {status: action.status});
        case 'REGISTER_USER_ERROR': return Object.assign(...state, {status: action.status});
        default: return state;
    }
}