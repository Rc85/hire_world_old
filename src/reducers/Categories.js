const categoriesInitialState = {
    status: null,
    categories: null
}

export const Categories = (state = categoriesInitialState, action) => {
    switch(action.type) {
        case 'ADD_CATEGORY_SUCCESS':
        case 'STORE_CATEGORIES':
            return Object.assign(...state, {status: action.status, categories: action.categories});
        case 'ADD_CATEGORY_BEGIN':
        case 'ADD_CATEGORY_FAIL':
        case 'ADD_CATEGORY_ERROR':
        case 'STORE_CATEGORIES_ERROR':
            return Object.assign(...state, {status: action.status});
        default: return state;
    }
}