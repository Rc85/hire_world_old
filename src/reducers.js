import { combineReducers } from 'redux';
import { ToggleMenu } from './reducers/Togglers';
import { Login } from './reducers/Auths';
import { Sectors } from './reducers/Sectors';
import { Confirmation } from './reducers/Confirmation';

export const reducers = combineReducers({
    ToggleMenu, Login, Sectors, Confirmation
});