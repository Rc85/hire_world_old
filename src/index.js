import 'babel-polyfill';
import 'normalize.css';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import './styles/vendor/bootstrap.css';
import './styles/vendor/bootstrap-grid.css';
import './styles/vendor/bootstrap-reboot.css';
import './styles/index.css';
import App from './App';
import { Provider } from 'react-redux';
import { reducers } from './reducers';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

const store = createStore(reducers, applyMiddleware(thunk));
store.subscribe(() => { console.log(store.getState()) });

ReactDOM.render(
    <Provider store={store}>
        <Router basename='/'>
            <App />
        </Router>
    </Provider>,
    document.getElementById('root')
);