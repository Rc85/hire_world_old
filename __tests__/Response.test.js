import React from 'react';
import Enzyme, { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router';
import { reducers } from '../src/reducers';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import App from '../src/App';

const store = createStore(reducers, applyMiddleware(thunk));

Enzyme.configure({adapter: new Adapter()});

describe('Response page', () => {
    test('snapshot', () => {
        const wrapper = shallow(
            <Provider store={store}><MemoryRouter initialEntries={[{pathname: '/some-unknown-route', key: 'response'}]}><App /></MemoryRouter></Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});