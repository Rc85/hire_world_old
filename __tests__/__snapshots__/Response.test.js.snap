// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response page snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "response",
        "pathname": "/some-unknown-route",
      },
    ]
  }
>
  <withRouter(Connect(App)) />
</MemoryRouter>
`;
